practice_app/pages/index.js

```diff
@@ -187,6 +192,30 @@ export default function Home({ context }) {
         }
     }
 
+    // This function returns the translated version of the input, to the language spoken in user's location
+    // current functionality : source_lang=english, target_lang=ip_dependent
+    const getTranslation = async () => {
+        // Learn client's ip adress using a third party service
+        const ipinfo = await axios.get('https://json.geoiplookup.io/')
+        // Provide client's ip & original text to our api
+        const oText = document.getElementById('originalText').value
+        const sourceLang = document.getElementById('sourceLanguage').value
+        const { data } = await axios.get(
+            encodeURI(
+                `api/getTranslation?countryCode=${ipinfo.data.country_code}&text=${encodeURIComponent(
+                    oText,
+                )}&sl=${sourceLang}`,
+            ),
+        )
```

Comment:
I believe there is no need for `encodeURI` or `encodeURIComponent`. Can you try it without?

Reply:
This is required to avoid "ERR_UNESCAPED_CHARACTERS" when the input is Chinese, Russian, etc. I think you mean one of them must be unnecessary, however, it just works with both of them.

Reply:
Oh okay I didn't expect non-latin languages to cause this error. Thanks